### PR TITLE
Add googletest & more!

### PR DESCRIPTION
**Pull request by ehsannas**
* Added googletest.
* Added "-spirv" option to hctbuild.
* Added "spirv" and "spirv_only" targets for hcttest.
* Added a sample unit test for the ModuleBuilder class.

Running `hctbuild` does *not* build googletest and SPIRV unit tests.
Running `hcttest` does *not* run SPIRV unit tests.

Running `hctbuild -spirv` does build googletest and SPIRV unit tests.
Running `hcttest spirv` will run ALL tests including SPIRV unit tests.
Running `hcttest spirv_only` will ONLY run SPIRV unit tests.

In short, the default behavior of hctbuild and hcttest is not changed.

Passes all tests.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -611,7 +613,7 @@ else()
   endif()
 endif()
 
-if(LLVM_INCLUDE_TESTS)
+if(LLVM_INCLUDE_TESTS OR ENABLE_SPIRV_GENERATION_FLOW)
```
Comment (antiagainst) [inline CMakeLists.txt]: Do we need really it here?

File: cmake/modules/AddLLVM.cmake [hunk 2]
```diff
@@ -754,6 +754,7 @@ function(add_unittest test_suite test_name)
   endif()
 
   include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googletest/include)
+  include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googlemock/include)
```
Comment (antiagainst) [inline cmake/modules/AddLLVM.cmake]: Should mark with `# SPIRV change`.

File: CMakeLists.txt [hunk 3]
```diff
@@ -84,6 +84,8 @@ option(HLSL_ENABLE_FIXED_VER "Sets up fixed version information." OFF) # HLSL Ch
 option(HLSL_ENABLE_ANALYZE "Enables compiler analysis during compilation." OFF) # HLSL Change
 option(HLSL_OPTIONAL_PROJS_IN_DEFAULT "Include optional projects in default build target." OFF) # HLSL Change
 
+option(ENABLE_SPIRV_GENERATION_FLOW "Enables the HLSL to SPIR-V conversion flow." OFF) # SPIRV Change
```
Comment (antiagainst) [inline CMakeLists.txt]: I feel this is a little bit too long. Maybe just `ENABLE_SPIRV_CODEGEN`?

File: tools/clang/unittests/SPIRV/ModuleBuilderTest.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+TEST(ValidateModuleBuilder, ValidateModuleWithHeaderOnly) {
+  clang::spirv::SPIRVContext context;
+  clang::spirv::ModuleBuilder builder(&context);
+  builder.BeginModule();
+  builder.EndModule();
+  std::vector<uint32_t> spvModule = builder.TakeModule();
+  // At the very least, running BeginModule() and EndModule() should
+  // create the SPIR-V Header. The header is exactly 5 words long.
+  EXPECT_NE(spvModule.size(), 0u);
+  EXPECT_EQ(spvModule.size(), 5u);
```
Comment (antiagainst) [inline tools/clang/unittests/SPIRV/ModuleBuilderTest.cpp]: I think we can just remove the first `EXPECT_NE` check. The second suffices. It would be nice if we can check all the words:

```cpp
ASSERT_THAT(spvModule, ElementsAre(spv::MagicNumber, spv::Version, ~0, 0, 0));
```

File: tools/clang/unittests/SPIRV/ModuleBuilderTest.cpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "clang/SPIRV/ModuleBuilder.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+TEST(ValidateModuleBuilder, ValidateModuleWithHeaderOnly) {
```
Comment (antiagainst) [inline tools/clang/unittests/SPIRV/ModuleBuilderTest.cpp]: Yay!

File: utils/hct/hctbuild.cmd [hunk 6]
```diff
@@ -100,6 +100,11 @@ if "%1"=="-vs2017" (
   set BUILD_GENERATOR=Visual Studio 15 2017
   shift /1
 )
+if "%1"=="-spirv" (
```
Comment (antiagainst) [inline utils/hct/hctbuild.cmd]: Should mark with `SPIRV change`

File: utils/hct/hcttest.cmd [hunk 7]
```diff
@@ -9,6 +9,8 @@ set TEST_CLANG=1
 set TEST_EXEC=1
 set TEST_CLANG_VERIF=0
 set TEST_EXTRAS=1
+set TEST_SPIRV=0
```
Comment (antiagainst) [inline utils/hct/hcttest.cmd]: Should mark with `SPIRV change`

File: utils/hct/hcttest.cmd [hunk 8]
```diff
@@ -20,6 +22,17 @@ if "%1"=="clean" (
   shift /1
 )
 
+if "%1"=="spirv" (
```
Comment (antiagainst) [inline utils/hct/hcttest.cmd]: Should mark with `SPIRV change`

File: utils/hct/hcttest.cmd [hunk 9]
```diff
@@ -98,6 +111,22 @@ if "%TEST_CLEAN%"=="1" (
   exit /b 0
 )
 
+if "%TEST_SPIRV%"=="1" (
```
Comment (antiagainst) [inline utils/hct/hcttest.cmd]: Should mark with `SPIRV change`
